File: inject-java/src/test/groovy/io/micronaut/inject/configproperties/MyHibernateConfig2.java
package io.micronaut.inject.configproperties;

import io.micronaut.context.annotation.ConfigurationProperties;
import io.micronaut.core.convert.format.MapFormat;
import io.micronaut.core.naming.conventions.StringConvention;

import java.util.Map;

@ConfigurationProperties("jpa")
public class MyHibernateConfig2 {
    private Map<String, String> properties;

    public Map<String, String> getProperties() {
        return properties;
    }

    public void setProperties(
            @MapFormat(transformation = MapFormat.MapTransformation.FLAT) Map<String, String> properties) {
        this.properties = properties;
    }
}

File: inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/inject/configproperties/MyHibernateConfig2.kt
package io.micronaut.kotlin.processing.inject.configproperties

import io.micronaut.context.annotation.ConfigurationProperties
import io.micronaut.core.convert.format.MapFormat

@ConfigurationProperties("jpa")
class MyHibernateConfig2 {

    @MapFormat(transformation = MapFormat.MapTransformation.FLAT)
    var properties: Map<String, String>? = null
}
